android/SmartLock/app/src/main/java/org/opendds/smartlock/OpenDdsBridge.java

```diff
@@ -32,39 +31,30 @@
 import SmartLock.lock_t;
 import SmartLock.vec2;
 
-public class OpenDdsService extends Service {
+public class OpenDdsBridge extends Thread {
+    private final static String LOG_TAG = "SmartLock_OpenDDS_Bridge";
 
-    final static String LOG_TAG = "SmartLock_OpenDDS_Service";
+    // need a persistent reference to avoid GC
+    private static DomainParticipantFactory participantFactory;
+    private static DomainParticipant participant = null;
+    private static ParticipantLocationListener locationListener = null;
+    private static DataWriter dw = null;
 
-    final static String LOCK_UPDATE_MESSAGE = "LockUpdateMessage";
-    final static String LOCK_STATUS_DATA = "LockStatus";
+    private  DomainParticipantQos participantQos;
 
     public boolean secure = false;
-
-    public DomainParticipantFactory participantFactory;
-    public DomainParticipantQos participantQos;
-
     private final int DOMAIN = 42;
+    private String[] groups;
 
-    private final IBinder binder = new OpenDdsBinder();
-
-    private static DomainParticipant participant = null;
-
-    private MainActivity activity = null;
-
-    protected MainActivity getActivity() {
-        return activity;
-    }
+    private final MainActivity activity;
+    private final Context context;
 
-    protected void setActivity(MainActivity activity) {
-        Log.i(LOG_TAG, "Activity is " + activity.toString());
+    public OpenDdsBridge(MainActivity activity) {
         this.activity = activity;
+        this.context = activity.getApplicationContext();
     }
 
-    // need a persistent reference to datawriter to avoid GC
-    private static DataWriter dw = null;
-
-    public boolean updateLockState (SmartLockStatus lockState)
+    public void updateLockState (SmartLockStatus lockState)
```

Comment:
```suggestion
    public void updateLockState(SmartLockStatus lockState)
```

There's a bunch of these (mostly in code that's not changed in this PR)